File: app/src/main/java/com/example/mymessenger/MainActivity2.java
package com.example.mymessenger;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.view.MenuItem;

import com.example.mymessenger.databinding.ActivityMain2Binding;
import com.google.android.material.navigation.NavigationBarView;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.database.FirebaseDatabase;

public class MainActivity2 extends AppCompatActivity {
    ActivityMain2Binding binding;
    HomeFragment homeFragment = new HomeFragment();
    MainFragment mainFragment = new MainFragment();
    SettingFragment settingFragment = new SettingFragment();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        binding = ActivityMain2Binding.inflate(getLayoutInflater());
        super.onCreate(savedInstanceState);
        setContentView(binding.getRoot());
        getSupportActionBar().hide();
        FirebaseAuth mAuth = FirebaseAuth.getInstance();
        FirebaseDatabase database = FirebaseDatabase.getInstance();
        FirebaseUser fUser = mAuth.getCurrentUser();
        if(fUser==null){
            startActivity(new Intent(this,AskActivity.class));
            finishAffinity();
        }
        replace(homeFragment);
        binding.bottomNavigationView.setOnItemSelectedListener(new NavigationBarView.OnItemSelectedListener() {
            @SuppressLint("NonConstantResourceId")
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                switch (item.getItemId()){
                    case R.id.homeFragment: replace(homeFragment);
                    break;
                    case R.id.mainFragment:replace(mainFragment);
                    break;
                    case R.id.settingFragment:replace(settingFragment);
                    break;
                    default:replace(homeFragment);
                }
                return true;
            }
        });

    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        startActivity(new Intent(MainActivity2.this,AskActivity.class));
        finishAffinity();
    }

    private void replace(Fragment fragment) {
        FragmentManager fragmentManager=getSupportFragmentManager();
        FragmentTransaction transaction=fragmentManager.beginTransaction();
        transaction.replace(R.id.fragment_container,fragment);
        transaction.commit();
    }

}

File: app/src/main/java/com/example/mymessenger/MainActivity2.kt
package com.example.mymessenger

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.navigation.findNavController
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.setupWithNavController
import com.example.mymessenger.R.id.homeFragment
import com.example.mymessenger.databinding.ActivityMain2Binding
import com.example.mymessenger.template.userVerification
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import com.google.firebase.database.FirebaseDatabase
import com.ismaeldivita.chipnavigation.ChipNavigationBar

class MainActivity2 : AppCompatActivity() {
    private lateinit var binding: ActivityMain2Binding
    private lateinit var mAuth: FirebaseAuth
    private lateinit var database: FirebaseDatabase
    private lateinit var firebaseUser: FirebaseUser
    private val homeFragment = HomeFragment()
    private val mainFragment = MainFragment()
    private val settingFragment = SettingFragment()
    override fun onCreate(savedInstanceState: Bundle?) {
        binding = ActivityMain2Binding.inflate(layoutInflater)
        super.onCreate(savedInstanceState)
        setContentView(binding.root)
        supportActionBar?.hide()
        mAuth = FirebaseAuth.getInstance()
        database = FirebaseDatabase.getInstance()
        firebaseUser = mAuth?.currentUser!!
        val verify = userVerification(firebaseUser, database, Intent(this, AskActivity::class.java))
        verify.setContext(this)
        verify.verifyUser()
        replace(homeFragment)


        binding.bottomNavigationView.setOnItemSelectedListener {
            when (it.itemId) {
                R.id.homeFragment -> replace(homeFragment)
                R.id.mainFragment -> replace(mainFragment)
                R.id.settingFragment -> replace(settingFragment)
            }
            true
        }


    }

    fun replace(fragment: Fragment) {
        val transaction = supportFragmentManager.beginTransaction()
        transaction.replace(R.id.fragment_container, fragment)
        transaction.commit()
    }

    override fun onBackPressed() {
        super.onBackPressed()
        startActivity(Intent(this@MainActivity2, AskActivity::class.java))
        finishAffinity()
    }
}